File: c/chapitre_4/exo43Fils.c
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>

#define BUF 256 

int main(int argc , char **argv )
{
    int fd_read ,fd_write ;
    char message[BUF] ;

    // Verification des argument
    if ( argc != 3 )
    {
        fprintf(stderr , "Usage : %s <pipe[0]>  <pipe[1]", argv[0] ) ;
        exit(0);
    }

    // Recuperation descripteur 
    
    fd_read =  atoi(argv[1] ) ;
    fd_write = atoi(argv[2]);

    close(fd_write);

    if ( read(fd_read , message , BUF ) == -1 )
    {
        fprintf(stderr , "Erreur : read\n") ;
        exit(1);
    }
    
    // affichage du message 

    printf("Fils Message recu : %s" , message ) ;

    close(fd_read) ;

    return (0);
}

File: python/chapitre_4/exo43Fils.py
import os
import sys

BUF = 256

def main():
    # Verification des argument
    if len(sys.argv) != 3:
        print(f"Usage : {sys.argv[0]} <pipe[0]>  <pipe[1]>", file=sys.stderr)
        sys.exit(0)

    # Recuperation descripteur
    fd_read  = int(sys.argv[1])
    fd_write = int(sys.argv[2])

    os.close(fd_write)

    message = os.read(fd_read, BUF)
    if not message:
        print("Erreur : read", file=sys.stderr)
        sys.exit(1)

    # affichage du message
    print(f"Fils Message recu : {message.decode()}")

    os.close(fd_read)

    return 0

main()